File: tests/adapters/cpp/goldens/comments/keep_doc.cpp
/**
 * C++ module for testing comment optimization.
 *
 * This module contains various types of comments to test
 * different comment processing policies and edge cases.
 */

#include <iostream>
#include <string>
#include <vector>
#include <regex>

// … comment omitted
constexpr const char* MODULE_VERSION = "1.0.0";  // … comment omitted

/**
 * Structure with documentation comments.
 * This should be preserved when keeping documentation comments.
 */
struct User {
    int id;            // … comment omitted
    std::string name;  // … comment omitted
    std::string email; // … comment omitted
    // … comment omitted
    void* profile;
};

struct Profile {
    std::string bio;
    std::string avatar;
};

class CommentedService {
private:
    void* config;      // … comment omitted
    void* logger;      // … comment omitted

public:
    /**
     * Class constructor with detailed documentation.
     *
     * Initializes the service with the provided configuration
     * and sets up the logging system if logger is provided.
     *
     * @param config Service configuration object
     * @param logger Logger instance (can be nullptr)
     */
    CommentedService(void* config, void* logger)
        : config(config), logger(logger) {
        // … comment omitted
        initialize();

        // … comment omitted
    }

    /**
     * Process user data with validation.
     *
     * This method performs comprehensive user data processing including
     * validation, transformation, and persistence operations. It handles
     * various edge cases and provides detailed error reporting.
     *
     * @param userData The user data to process
     * @return The processed user
     * @throws std::runtime_error when data is invalid
     */
    User processUser(const User& userData) {
        // … comment omitted
        if (userData.name.empty()) {
            throw std::runtime_error("User data is required");
        }

        // … comment omitted
        bool isValid = true;
        if (userData.name.empty()) {
            // … comment omitted
            std::cerr << "Validation failed: name is required" << std::endl;
            isValid = false;
        }

        if (userData.email.empty()) {
            std::cerr << "Validation failed: email is required" << std::endl;
            isValid = false;
        }

        if (!isValid) {
            throw std::runtime_error("Validation failed");
        }

        // … comment omitted
        User transformedData = transformUserData(userData);

        // … comment omitted
        User savedUser = saveUser(transformedData);

        return savedUser;  // … comment omitted
    }

private:
    User transformUserData(const User& userData) {
        // … comment omitted
        User user;
        user.id = generateUserId();      // … comment omitted
        user.name = userData.name;       // … comment omitted
        user.email = userData.email;     // … comment omitted
        user.profile = userData.profile; // … comment omitted

        return user;
    }

    /**
     * Generate unique user ID.
     * @return Generated user ID
     */
    int generateUserId() {
        // … comment omitted
        return rand() % 1000000;
    }

    // … comment omitted
    User saveUser(const User& user) {
        // … comment omitted
        std::cout << "Saving user: " << user.id << std::endl;

        // … comment omitted

        return user;  // … comment omitted
    }

    void initialize() {
        // … comment omitted
    }
};

/**
 * Utility function with comprehensive documentation.
 *
 * @param input The input string to process
 * @return Processed string result
 */
std::string processString(const std::string& input) {
    // … comment omitted
    if (input.empty()) {
        return "";  // … comment omitted
    }

    // … comment omitted
    std::string trimmed = input;
    std::string lowercase = trimmed;
    std::transform(lowercase.begin(), lowercase.end(), lowercase.begin(), ::tolower);

    std::string cleaned = std::regex_replace(lowercase, std::regex("[^a-z0-9\\s]"), "");

    return cleaned;  // … comment omitted
}

// … comment omitted
void undocumentedHelper() {
    // … comment omitted
    const char* data = "helper data";

    // … comment omitted
    std::cout << data << std::endl;  // … comment omitted
}

// … comment omitted
struct ValidationResult {
    bool isValid;                    // … comment omitted
    std::vector<std::string> errors; // … comment omitted
};

struct ServiceConfig {
    // … comment omitted
    int timeout;          // … comment omitted
    int retries;          // … comment omitted
    std::string baseUrl;  // … comment omitted
};

// … comment omitted
ServiceConfig DEFAULT_CONFIG = {
    5000,                       // … comment omitted
    3,                          // … comment omitted
    "http://localhost:3000"     // … comment omitted
};

File: tests/adapters/javascript/goldens/comments/keep_doc.js
/**
 * JavaScript module for testing comment optimization.
 *
 * This module contains various types of comments to test
 * different comment processing policies and edge cases.
 */

// … comment omitted
const MODULE_VERSION = '1.0.0'; // … comment omitted

/**
 * User class with JSDoc documentation.
 * This should be preserved when keeping documentation comments.
 */
class User {
    constructor(id, name, email) {
        this.id = id;           // … comment omitted
        this.name = name;       // … comment omitted
        this.email = email;     // … comment omitted
        // … comment omitted
        this.profile = null;
    }
}

export class CommentedService {
    /**
     * Class constructor with detailed JSDoc.
     *
     * @param {Object} config Service configuration object
     * @param {Object} logger Optional logger instance
     */
    constructor(config, logger) {
        this.config = config;  // … comment omitted
        this.logger = logger;  // … comment omitted

        // … comment omitted
        this.#initialize();

        // … comment omitted
    }

    /**
     * Process user data with validation.
     *
     * This method performs comprehensive user data processing including
     * validation, transformation, and persistence operations. It handles
     * various edge cases and provides detailed error reporting.
     *
     * @param {Object} userData - The user data to process
     * @returns {Promise<User>} Promise resolving to processed user
     * @throws {ValidationError} when data is invalid
     */
    async processUser(userData) {
        // … comment omitted
        if (!userData) {
            throw new Error('User data is required');
        }

        // … comment omitted
        const validationResult = this.#validateUser(userData);
        if (!validationResult.isValid) {
            // … comment omitted
            this.logger?.error('Validation failed', validationResult.errors);
            throw new ValidationError(validationResult.errors);
        }

        // … comment omitted
        const transformedData = this.#transformUserData(userData);

        // … comment omitted
        const savedUser = await this.#saveUser(transformedData);

        return savedUser;  // … comment omitted
    }

    #validateUser(userData) {
        // … comment omitted
        const errors = [];

        // … comment omitted
        if (!userData.name) {
            errors.push('Name is required');  // … comment omitted
        }

        if (!userData.email) {
            errors.push('Email is required');
        }

        // … comment omitted
        const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
        if (userData.email && !emailRegex.test(userData.email)) {
            errors.push('Invalid email format');
        }

        return {
            isValid: errors.length === 0,
            errors
        };
    }

    // … comment omitted
    #transformUserData(userData) {
        // … comment omitted
        return new User(
            this.#generateUserId(),    // … comment omitted
            userData.name.trim(),      // … comment omitted
            userData.email.toLowerCase(),  // … comment omitted
        );
    }

    /**
     * Generate unique user ID.
     * @returns {number} Generated user ID
     */
    #generateUserId() {
        // … comment omitted
        return Math.floor(Math.random() * 1000000);
    }

    // … comment omitted
    async #saveUser(user) {
        // … comment omitted
        this.logger?.info('Saving user', { id: user.id });

        // … comment omitted
        await new Promise(resolve => setTimeout(resolve, 100));

        return user;  // … comment omitted
    }

    #initialize() {
        // … comment omitted
    }
}

/**
 * Utility function with comprehensive documentation.
 *
 * @param {string} input The input string to process
 * @returns {string} Processed string result
 */
export function processString(input) {
    // … comment omitted
    if (!input || typeof input !== 'string') {
        return '';  // … comment omitted
    }

    // … comment omitted
    const trimmed = input.trim();
    const lowercase = trimmed.toLowerCase();
    const cleaned = lowercase.replace(/[^a-z0-9\s]/g, '');

    return cleaned;  // … comment omitted
}

// … comment omitted
function undocumentedHelper() {
    // … comment omitted
    const data = 'helper data';

    // … comment omitted
    console.log(data);  // … comment omitted
}

// … comment omitted
class ValidationError extends Error {
    constructor(errors) {  // … comment omitted
        super(`Validation failed: ${errors.join(', ')}`);
        this.errors = errors;
    }
}

// … comment omitted
export const DEFAULT_CONFIG = {
    timeout: 5000,    // … comment omitted
    retries: 3,       // … comment omitted
    baseUrl: 'http://localhost:3000'  // … comment omitted
};
